src/Sdk/StrixMusic.Sdk.Tests/MetadataScanner/AudioMetadataScanner.Tests.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using StrixMusic.Sdk.FileMetadata;
+using StrixMusic.Sdk.FileMetadata.Scanners;
+using StrixMusic.Sdk.Tests.Mock.FileSystem;
+
+namespace StrixMusic.Sdk.Tests.Services.MetadataScanner
+{
+    [TestClass]
+    public class AudioMetadataScannerTests
+    {
+        [DataRow("MultipleSongArtists.mp3", MetadataScanTypes.TagLib)]
+        [DataRow("MultipleSongArtists2.mp3", MetadataScanTypes.TagLib)]
+        [TestMethod, Timeout(1000)]
+        public async Task AlbumArtist_And_MultipleTrackArtist(string fileName, MetadataScanTypes scanTypes)
+        {
+            var scanner = new AudioMetadataScanner(degreesOfParallelism: 2);
+            scanner.ScanTypes = scanTypes;
+            scanner.CacheFolder = new SystemFolderData(Path.GetTempPath()); // TODO: https://github.com/Arlodotexe/strix-music/issues/160
+
+            var filePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, @"MetadataScanner\Samples", fileName);
+            var file = new SystemFileData(filePath);
+
+            var metadata = await scanner.ScanMusicFiles(new[] { file });
+            Assert.IsTrue(metadata.Count() > 0);
+
+            foreach (var item in metadata)
+            {
+                Assert.IsNotNull(item.AlbumArtistMetadata);
+                Assert.IsTrue(item.AlbumArtistMetadata.Count() > 0);
+
+                Assert.IsNotNull(item.TrackArtistMetadata);
+                Assert.IsTrue(item.TrackArtistMetadata.Count() > 1);
+            }
+        }
+
+        [DataRow("NoMetadata.mp3", MetadataScanTypes.TagLib)]
+        [TestMethod, Timeout(1000)]
+        public async Task NoMetadata(string fileName, MetadataScanTypes scanTypes)
+        {
+            var scanner = new AudioMetadataScanner(degreesOfParallelism: 2);
+            scanner.ScanTypes = scanTypes;
+            scanner.CacheFolder = new SystemFolderData(Path.GetTempPath()); // TODO: https://github.com/Arlodotexe/strix-music/issues/160
+
+            var filePath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, @"MetadataScanner\Samples", fileName);
+            var file = new SystemFileData(filePath);
+
+            var metadata = await scanner.ScanMusicFiles(new[] { file });
+            Assert.IsTrue(metadata.Count() > 0);
+
+            foreach (var item in metadata)
```

Comment:
You dont had to check all properties on no metadata but its nice 🙂

Reply:
I did have to - I wanted a quick and repeatable proof that feeding the scanner a file with no data works exactly as expected.